api/server_api.go

```diff
@@ -95,14 +116,19 @@ func (s *Server) HandleWs(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	serverIpNet, err := s.getServerIpNet(conn.LocalAddr().String())
+	if err != nil {
+		panic(err)
+	}
+
 	sessionIdQuery := r.URL.Query().Get(URLQuerySessionIDKeyword)
 	switch sessionIdQuery {
 	case "":
 		// creating a new URL compatible session ID
 		newSessionIdRaw := uuid.New().String()
 		sessionIdUrlCompatible := base64.RawURLEncoding.EncodeToString([]byte(newSessionIdRaw))
 
-		session := NewSession(conn, sessionIdUrlCompatible, s.GameManager, s.SessionManager)
+		session := NewSession(conn, sessionIdUrlCompatible, s.GameManager, s.SessionManager, serverIpNet, s.Db)
```

Comment:
I think there is no point in passing the db and serverIPNet to session. By holding a reference of db and serverIpNet, session and servers will be tightly coupled.

Reply:
I'd need that IP and db pointer to execute the sql operations.

I'm thinking what if we just pass a pointer to `Server` instead of the fields below
```go
GameManager    *GameManager
SessionManager *SessionManager
Db             *sql.DB
ServerIPNet    net.IPNet
```

Reply:
we can have an `AnalyticsManager` that is singelton that holds the db reference. Whenever needed, we can call it like:
```go
server.AnalyticsManager.incrementGameCount()
```

Reply:
I like this idea but this can be done in a simpler way.
Queries struct produced by sqlc already has all the methods with proper namings.

Let's just create a pointer to that and pass it to the server, that would cover all the needs (Even for the future that we might have other sql operations)